### PR TITLE
Lo l2 sputtering

### PR DESCRIPTION
**Pull request by greglucas**
# Change Summary

## Overview

The final commit is the sputtering correction (rebased onto the single-species update).

For now, I've decided for a basic implementation of the equations and pass-through of the oxygen dataset. I will need to think about this some more in terms of how we actually want to handle the h/o datasets. The major piece here is that this adds the equations and tests to calculate what the sputtering correction terms should be.

part of: #1894

### CODE REVIEW

File: imap_processing/lo/l2/lo_l2.py [hunk 1]
```diff
@@ -728,6 +738,91 @@ def calculate_backgrounds(dataset: xr.Dataset) -> xr.Dataset:
...
+
+    # Equation 10
+    j_o_prime_var = (
+        o_small_dataset["ena_intensity_stat_uncert"] ** 2
+        + o_small_dataset["bg_intensity_stat_uncert"] ** 2
+    )
+
+    # NOTE: From table 2 of the mapping document, for energy level 5 and 6
+    sputter_correction_factor = xr.DataArray(
+        [0.15, 0.01], dims=["energy"], coords={"energy": [4, 5]}
```
Comment (tmplummer) [inline imap_processing/lo/l2/lo_l2.py]: Any idea if these correction factors are likely to change? Do they need to be in an ancillary calibration file?

File: imap_processing/lo/l2/lo_l2.py [hunk 2]
```diff
@@ -728,6 +738,91 @@ def calculate_backgrounds(dataset: xr.Dataset) -> xr.Dataset:
...
+        and uncertainties.
+
+    Returns
+    -------
+    xr.Dataset
+        Dataset with calculated sputtering-corrected intensities and their
+        uncertainties for hydrogen and oxygen.
+    """
+    logger.info("Applying sputtering corrections to oxygen intensities")
+    # Only apply sputtering correction to esa levels 5 and 6
```
Comment (tmplummer) [inline imap_processing/lo/l2/lo_l2.py]: You might consider stating that levels 5 and 6 correspond to indices 4 and 5.

I would also recommend storing the energy indices in a variable rather than hard coding them everywhere.

File: imap_processing/lo/l2/lo_l2.py (hunk 1 above)
Comment (tmplummer) [inline imap_processing/lo/l2/lo_l2.py]: The more I use xarray, the more I like it! I didn't know that you could provide `coords` in a mapping like this.

File: imap_processing/tests/lo/test_lo_l2.py
Comment (tmplummer) [inline imap_processing/tests/lo/test_lo_l2.py:1]: Wow, that is some very thorough testing.

**Review by tmplummer — APPROVED** [review]
LGTM. Just a couple nits that you can decide what to do with.

File: imap_processing/lo/l2/lo_l2.py (hunk 1 above)
Comment (greglucas) [reply imap_processing/lo/l2/lo_l2.py]: I don't think so because these are from an external paper reference, but I don't know for sure. For now, I'm taking the tact of putting these factors from the document as hard-coded arrays because they are from the table, but that it wouldn't be hard to switch in the future if we do find we need to do that.

File: imap_processing/tests/lo/test_lo_l2.py
Comment (greglucas) [reply imap_processing/tests/lo/test_lo_l2.py:1]: Claude is really helping me out on the tests! Honestly, it is a bit duplicative for my liking, so maybe in the future we could come back in and clean things up, but it did quite a good job with coverage and edge cases and much faster than what I could write.

File: imap_processing/lo/l2/lo_l2.py (hunk 2 above)
Comment (greglucas) [reply imap_processing/lo/l2/lo_l2.py]: Good call, updated this!

**Comment by greglucas** [comment]
Merging over the known failure again to not have to rerun the tests.
